### PR TITLE
Issue #1109: feature-awesomebar: Connecting a concept-awesomebar with…

### PR DESCRIPTION
**Pull request by pocmo**
Depends on `browser-awesomebar` (See PR #1260).

~~The build will fail until PR #1260 is in.~~

~~In addition to that I still need to write some tests. However I wanted to put this up for review already.~~

### CODE REVIEW

**Comment by pocmo** [comment]
> The build will fail until PR #1260 is in.

Heh, fooled by `concept` components. Of course this does not depend on the actual implementation. :)

**Comment by pocmo** [comment]
Added some tests.

File: components/feature/awesomebar/src/main/java/mozilla/components/feature/awesomebar/provider/SearchSuggestionProvider.kt [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+/**
+ * A [AwesomeBar.SuggestionProvider] implementation that provides a suggestion containing search engine suggestions (as
+ * chips) from the passed in [SearchEngine].
+ */
+class SearchSuggestionProvider(
+    private val searchEngine: SearchEngine,
+    private val searchUseCase: SearchUseCases.DefaultSearchUrlUseCase
+) : AwesomeBar.SuggestionProvider {
+    private val client = if (searchEngine.canProvideSearchSuggestions) {
+        SearchSuggestionClient(searchEngine, { url -> fetch(url) })
```
Comment (Amejia481) [inline components/feature/awesomebar/src/main/java/mozilla/components/feature/awesomebar/provider/SearchSuggestionProvider.kt]: nit: As the lambda is the last parameter  you could pull it the  outside of the parenthesis.
```kotlin
SearchSuggestionClient(searchEngine){ url -> 
    fetch(url) 
} 
```

Comment (pocmo) [reply components/feature/awesomebar/src/main/java/mozilla/components/feature/awesomebar/provider/SearchSuggestionProvider.kt]: Oh yeah, that makes sense. I'll update the patch.

**Comment by pocmo** [comment]
Updated!

**Comment by Amejia481** [comment]
```
There's a small format issue
> Task :ktlint
/build/android-components/components/feature/awesomebar/src/main/java/mozilla/components/feature/awesomebar/provider/SearchSuggestionProvider.kt:25:44: Unexpected spacing before ")"

```

**Pull request merged**